index.html

```diff
@@ -179,44 +194,34 @@
                         if (enemy.y < Crafty('PlayerCharacter').y) {
                             enemy.velocity().y = 100
                         } 
-                        if (-30 < enemy.x - Crafty('PlayerCharacter').x < 30 || -30 < enemy.y - Crafty('PlayerCharacter').y < 30)
-                            var villain = this;
```

Comment:
your problem was trying to capture `villain` like this, inside the callback to setTimeout (where `this` is undefined -- rmemeber the this inside of this issue?). You already have the enemy, so just use it.